### ISSUE TITLE
Display configured VCS author

### ISSUE THREAD

**Issue by nojhan** [nojhan/liquidprompt#785]
Add an optional segment displaying the currently configured author in a VCS repository, if any.

### Description

If a local author is configured in the repository, display a short version of their name.

If no local author is configured, then the global configuration applies and the user should already know it, so there is no need for a warning, thus no need to display the name.

This would need:

- a config file parser, which already exists in #784 for Git.
- a name shortener, because we don't want to display the whole author.

The name shortener can probably just display the initials, which should be enough in most cases, but a set of configurable methods may also be useful for complex cases:

- initials
- given name
- first n letters of the name
- e-mail prefix
- e-mail domain
- e-mail domain without TLD
- initials of e-mail prefix
- initials of all e-mails components
- any combination thereof?

### How will this be useful?

I regularly use two authors to differentiate personal and work handles. It happened more than once that I fresh clone some repo (across several computers); then forget to change the local author, ending in commiting with my personal account on a pro repo. Fixing such an error is cumbersome, hence the need to avoid it.

### Example prompt

`[:~/code/liquidprompt] feat/mem+* (noj) ±`
`[:~/code/liquidprompt] feat/remote+* ⭚origin(AB) ±`
`[:~/code/liquidprompt] feat/remote+* ⭚origin(n@n.n) ±`

**Comment by Rycieos** [comment]
This isn't a terrible idea. I personally wouldn't find it too useful, but I can see your use case.

If I may propose a better way to solve your problem; I separate my repos into two directories, `~/dev/` and `~/work/`. Personal stuff goes in `~/dev/`, and work stuff obviously in `~/work/`. I have a block at the end of my `~/.config/git/config` like this:
```ini
[includeIf "gitdir:~/work/**"]
    path = ~/work/.gitconfig/config
```

And `~/work/.gitconfig/config` contains:

```ini
[core]
    hooksPath = ~/work/.gitconfig/hooks
[commit]
    gpgsign = true
[user]
    email = me@work.domain
    signingkey = WORKGPGKEYID
```
You could have a separate name in there as well of course.

This means I don't need to remember to configure each repo; by putting it in `~/work/`, I get that config for free.

**Comment by nojhan** [comment]
I tried this approach and got converted :-)
I also took the habit of naming the remotes from the account (not just "origin"), and that helped avoid most of the problems I had.

Now, at least two users liked the idea, so I have mixed feelings about implementing this, since the name length/shortening can be an issue, from a user experience perspective.

**Comment by Rycieos** [comment]
> I have mixed feelings about implementing this, since the name length/shortening can be an issue, from a user experience perspective.

I agree. A famous blog post is [Falsehoods Programmers Believe About Names](https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/). The short version is that no matter what method you come up with to reduce a full name, it will be incorrect for at least one person.

That said, I think some data source like `_vcs_author_name()` and `_vcs_author_email()` is not a bad idea. I would either leave it out of the default theme, or have the feature disabled by default. I would leave the data unmodified; if a theme wanted to truncate in some way that would be up to the theme.
